src/main/java/uk/gov/hmcts/reform/finrem/caseorchestration/service/documentremoval/DocumentRemovalService.java

```diff
@@ -0,0 +1,37 @@
+package uk.gov.hmcts.reform.finrem.caseorchestration.service.documentremoval;
+
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.node.ArrayNode;
+import lombok.RequiredArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.stereotype.Service;
+
+import java.util.Iterator;
+import java.util.List;
+
+@Service
+@Slf4j
+@RequiredArgsConstructor
+public class DocumentRemovalService {
+
+    public void retrieveDocumentNodes(JsonNode root, List<JsonNode> documentNodes) {
```

Comment:
I think it would make a better API if you changed the argument to be FinremCaseData and returned the nodes. Why make clients do all the object mapper stuff?

`public List<JsonNode> documentNodes retrieveDocumentNodes(FinremCaseData caseData)`

Reply:
Given how this method uses recursion to traverse through the nested JSON structure, I would not be able to change the params of this method to FinRemCaseData Type. Similarly, the method works by adding to values to the list that is passed as a param on the recursion calls, as such it should remain as a method param rather than being a return type. 

I could potentially abstract the map from FinRemCaseData to JSON node, however, I'm not sure if this is worth abstracting.

Reply:
Had a chat with @so99y, just going to go down the root of creating an override method for multiple type handlers

Reply:
You can do this. The recursive method can be separate from your API method.